support/sier-codec/src/lib.rs

```diff
@@ -70,6 +134,7 @@ pub enum Error<'i> {
     TooFewBytes,
     TooManyBytes,
     InvalidUtf8(std::str::Utf8Error),
+    InvalidJson,
```

Comment:
We should include some extra information about what the error is. Otherwise it will be hard for the users to know what's wrong. Perhaps we should look into anyhow or error_chain or whatever new crate people are using for errors.

Reply:
I think we should start using `thiserror` in the whole lib. This means I should refactor the error handling everywhere. Maybe it would be better to do it in another PR? For this one I added one more type of error.

Reply:
Happy to use whatever error library is the current standard in Rust (they change all the time). Just make sure it doesn't have `std` requirements.

We don't want to have `MismatchTypeFromJson` long term, and removing a variant of an enum is a breaking change. Let's just keep `InvalidJson` for now, and create Json specific errors in that module once we integrate/use `thiserror`.